backend/app/views/functional.py

```diff
@@ -896,11 +901,54 @@ def generate(current_user):
                                              "url": url})
         return response.success({"output": "https://www.google.com/search?q=" + query}, Status.OK)
 
+    elif mode == "qa":
+        if not query:
+            return {"message": "Query required for question-answer mode."}, 400
+        prompt = "Answer the following question with fewer than 100 words. Question: " + query + ". Answer: "
+
+    elif mode == "contextual_qa":
+        if not context:
+            return {"message": "Context required for in-context question generation."}, 400
+        if not query:
+            return {"message": "Query required for in-context question generation."}, 400
+        
+        prompt = context_intent_qgen_prompt + '"' + context + '" INTENT: ' + query + '" --> QUESTIONS:'        
+
+    elif mode == "gen_questions":
+        if not context:
+            return {"message": "Context required for question generation."}, 400
+
+        prompt = context_qgen_prompt + '"' + context + '" --> QUESTIONS:'
+
+    elif mode == "summarize":
+        if not context:
+            return {"message": "Context required for summarization."}, 400
+        prompt = context + "... Please summarize the previous text, and only reply with the summary. Summary: "
+
+    elif mode == "summary_rag":
+        if not search_id:
+            return {"message": "URL for exporting search results not provided"}, 400
```

Comment:
This error message should probably say something like "search_id is required for summarizing search results."